kernel/src/actions/visitors.rs

```diff
@@ -196,20 +196,20 @@ impl RemoveVisitor {
 
         let size: Option<i64> = getters[5].get_opt(row_index, "remove.size")?;
 
-        // TODO(nick) stats are skipped in getters[6] and tags are skipped in getters[7]
+        // TODO(nick) tags are skipped in getters[6]
```

Comment:
Next PR: derive macro for basic visitors? That would eliminate the possibility that they get out of sync.

For fields that are themselves structs, we could use that struct's own visitor (after verifying that the first non-nullable sub-field is non-null, in case the field was also nullable).

The one annoyance is all those e.g. `"remove.xxx"` error message helpers -- we'd have to either derive `remove` from the struct name, or else tell the macro the field name to use.